Tests/SentryTests/Integrations/SentryPerformanceTrackerTests.swift

```diff
@@ -0,0 +1,306 @@
+        let span = spans[spanId]
+        
+        sut.finishSpan(spanId, with: .ok)
+        
+        let status = Dynamic(span).finishStatus as SentrySpanStatus?
+        
+        XCTAssertEqual(status!, .ok)
+        XCTAssertTrue(span!.isFinished)
+    }
+    
+    func testIsSpanAlive() {
+        let sut = fixture.getSut()
+        let spanId = startSpan(tracker: sut)
+        sut.pushActiveSpan(spanId)
+        XCTAssertTrue(sut.isSpanAlive(spanId))
+
+        let childId = startSpan(tracker: sut)
+        XCTAssertTrue(sut.isSpanAlive(spanId))
+        XCTAssertTrue(sut.isSpanAlive(childId))
+        
+        sut.finishSpan(childId)
+        XCTAssertTrue(sut.isSpanAlive(spanId))
+        XCTAssertFalse(sut.isSpanAlive(childId))
+        
+        sut.finishSpan(spanId)
+        XCTAssertFalse(sut.isSpanAlive(spanId))
+    }
+    
+    @available(tvOS 10.0, *)
+    @available(OSX 10.12, *)
+    @available(iOS 10.0, *)
+    func testStartSpanAsync() {
+        let sut = fixture.getSut()
+        let spanId = startSpan(tracker: sut)
+        sut.pushActiveSpan(spanId)
+        
+        let queue = DispatchQueue(label: "SentryPerformanceTrackerTests", attributes: [.concurrent, .initiallyInactive])
+        let group = DispatchGroup()
+        
+        for _ in 0 ..< 5_000 {
+            group.enter()
+            queue.async {
+                _ = self.startSpan(tracker: sut)
+                group.leave()
+            }
+        }
+        
+        queue.activate()
+        group.wait()
+                
+        let spans = getSpans(tracker: sut)
+        XCTAssertEqual(spans.count, 5_001)
+    }
+    
+    @available(tvOS 10.0, *)
+    @available(OSX 10.12, *)
+    @available(iOS 10.0, *)
+    func testStackAsync() {
+        let sut = fixture.getSut()
+        let spanId = startSpan(tracker: sut)
+        sut.pushActiveSpan(spanId)
+        
+        var queue = DispatchQueue(label: "SentryPerformanceTrackerTests", attributes: [.concurrent, .initiallyInactive])
+        var group = DispatchGroup()
+        
+        for _ in 0 ..< 5_000 {
+            group.enter()
+            queue.async {
+                let childId = self.startSpan(tracker: sut)
+                sut.pushActiveSpan(childId)
+                group.leave()
+            }
+        }
+        
+        queue.activate()
+        group.wait()
+        
+        var stack = getStack(tracker: sut)
+        XCTAssertEqual(5_001, stack.count)
+        
+        queue = DispatchQueue(label: "SentryPerformanceTrackerTests", attributes: [.concurrent, .initiallyInactive])
+        group = DispatchGroup()
+        
+        for _ in 0 ..< 5_000 {
+            group.enter()
+            queue.async {
+                sut.popActiveSpan()
+                group.leave()
+            }
+        }
+        
+        queue.activate()
+        group.wait()
+        
+        sut.popActiveSpan()
+        
+        stack = getStack(tracker: sut)
```

Comment:
`m`: I think it would be better to push and pop in parallel and to increase the number a bit. 
```suggestion
        let queue = DispatchQueue(label: "SentryPerformanceTrackerTests", attributes: [.concurrent, .initiallyInactive])
        let group = DispatchGroup()
        
        for _ in 0 ..< 50_000 {
            group.enter()
            queue.async {
                let childId = self.startSpan(tracker: sut)
                sut.pushActiveSpan(childId)
                sut.popActiveSpan()
                group.leave()
            }
        }
        
        queue.activate()
        group.wait()
        
        sut.popActiveSpan()
        
        let stack = getStack(tracker: sut)
```